content/articles/2019-06-26-ragg-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,109 @@
+
+Apart from the three file-based devices, ragg also provides a device that gives direct access to the raster buffer from R. This means that you can plot directly into a matrix of color values that you can then further process in R, should you wish.
+
+## Features
+While the graphic engine in R allows devices to turn off some features, such as the ability to draw holed polygons or raster images (i.e. drawing a raster image into the device), all modern devices are feature complete and so is the ones provided by ragg. That being said, ragg provides some features that sets itself
+apart from the raster devices already included with R.
+
+#### Performance
+ragg is faster than cairo, the standard anti-aliased device in R. Tests show that it is about twice as fast to render a moderately complex ggplot. See the
+[Performance](https://ragg.r-lib.org/articles/ragg_performance.html) vignette for more in-depth benchmarking.
+
+#### Anti-aliasing
+ragg is fully anti-aliased, as opposed to the cairo devices that only apply anti-aliasing to strokes and text (not fill). For a comparison of the look, see the [shape rendering](https://ragg.r-lib.org/articles/ragg_quality.html#shape-rendering) section of the Quality vignette for comparisons.
+
+#### Text rendering
+ragg provides high-quality rendering of rotated text, something that other raster devices struggle with. See the [text](https://ragg.r-lib.org/articles/ragg_quality.html#text) section of the Quality vignette for examples.
+
+#### Font access
+ragg has direct access to all your system fonts, without you having to do anything. All installed fonts on your system should be ready to use.
+
+#### System independent
+The rendering code in ragg is system independent and should be identical whether
+it has been rendered on Windows, macOS, or Linux. Only difference may be in the available fonts on the different systems.
+
+#### 16-bit support
+ragg has the possibility of writing 16-bit png and tiff files. As the color implementation is strictly 8-bit this does not have a huge impact, but could provide quality differences in certain situations (at the expense of larger files).
+
+## Example
+The ragg devices are used just like any other device, by starting them, running your plotting code, and turning them off (code below add some complexity in order to embed the created file in the page):
+
+```{r}
+library(ragg)
+library(ggplot2)
+
+file <- knitr::fig_path('.png')
+
+agg_png(file, width = 700, height = 500, units = 'px')
+ggplot(diamonds) + 
+  geom_bar(aes(color, fill = color)) + 
+  ggtitle("A fancy font") + 
+  theme(text = element_text(family = "Daubmark", size = 50))
+invisible(dev.off())
+
+knitr::include_graphics(file)
+```
+
+The use of the device that allows direct access from R is a little different. It returns a function when you open it and calling that function will return the current state of the device as a matrix of color values:
```

Comment:
I think you can drop this because it's not that important.

Reply:
the `egg_capture` use?

Reply:
Yeah